IoTuring/Entity/Deployments/Volume/Volume.py

```diff
@@ -101,12 +111,39 @@ def UpdateMac(self):
             KEY_STATE, EXTRA_KEY_ALERT_VOLUME, alert_volume, valueFormatterOptions=VALUEFORMATTEROPTIONS_PERCENTAGE_ROUND0)
         self.SetEntitySensorExtraAttribute(
             KEY_STATE, EXTRA_KEY_MUTED_OUTPUT, output_muted)
+        
+    def UpdateWindows(self):
+        with Volume.WindowsVolumeControl() as volume_control:
+            output_volume = int(volume_control.GetMasterVolumeLevelScalar() * 100)
+            output_muted = volume_control.GetMute()
+
+        self.SetEntitySensorValue(KEY_STATE, output_volume)
+        self.SetEntitySensorExtraAttribute(
+            KEY_STATE, EXTRA_KEY_OUTPUT_VOLUME, output_volume, valueFormatterOptions=VALUEFORMATTEROPTIONS_PERCENTAGE_ROUND0)
+        self.SetEntitySensorExtraAttribute(
+            KEY_STATE, EXTRA_KEY_MUTED_OUTPUT, output_muted)
+        
+    @classmethod
+    @contextmanager
+    def WindowsVolumeControl(cls):
+        """
+        Context manager to retrieve the main Windows volume control.  
+
+        Since these methods can be called in a subprocess, we need to wrap 
+        the COM calls in CoInitialize and CoUninitialize calls.
+
+        See https://github.com/AndreMiras/pycaw/issues/34#issuecomment-826107126
+        """
+        comtypes.CoInitialize()
+        devices = AudioUtilities.GetSpeakers()
+        interface = devices.Activate(
+            IAudioEndpointVolume._iid_, comtypes.CLSCTX_ALL, None)
+        yield interface.QueryInterface(IAudioEndpointVolume)
+        comtypes.CoUninitialize()
 
     @classmethod
     def CheckSystemSupport(cls):
         if OsD.IsLinux():
```

Comment:
Can you write some check here, so it shows a reason if the imports fails? Something like what's happening here: https://github.com/richibrics/IoTuring/blob/4253570b603a3ccfd3b90a24ed2d102cf683d18a/IoTuring/Entity/Deployments/ActiveWindow/ActiveWindow.py

Otherwise it seems nice, I will try it on my win pc this week.

Reply:
Ah, yeah, that does seem like a cleaner way of handling those imports.  Just pushed that change.